### PR TITLE
[search] Disabled launching in the constructor of TestSearchRequest.

### PR DESCRIPTION
**Pull request by mpimenov**
@ygorshenin PTAL

### CODE REVIEW

**Comment by ygorshenin** [comment]
LGTM

File: search/search_tests_support/test_search_request.cpp [hunk 1]
```diff
@@ -66,10 +71,18 @@ vector<search::Result> const & TestSearchRequest::Results() const
...
+  m_params.m_onStarted = bind(&TestSearchRequest::OnStarted, this);
+  m_params.m_onResults = bind(&TestSearchRequest::OnResults, this, _1);
+}
+
+void TestSearchRequest::SetCustomOnResults(SearchParams::TOnResults const & customOnResults)
 {
-  params.m_onStarted = bind(&TestSearchRequest::OnStarted, this);
-  params.m_onResults = bind(&TestSearchRequest::OnResults, this, _1);
+  m_params.m_onResults = [this, customOnResults](search::Results const & results) {
+    OnResults(results);
```
Comment (ygorshenin) [inline search/search_tests_support/test_search_request.cpp]: We do not need to call original OnResults here.

File: search/search_tests_support/test_search_request.cpp [hunk 2]
```diff
@@ -66,10 +71,18 @@ vector<search::Result> const & TestSearchRequest::Results() const
...
 }
 
-void TestSearchRequest::SetUpCallbacks(SearchParams & params)
+void TestSearchRequest::SetUpCallbacks()
+{
+  m_params.m_onStarted = bind(&TestSearchRequest::OnStarted, this);
+  m_params.m_onResults = bind(&TestSearchRequest::OnResults, this, _1);
+}
+
+void TestSearchRequest::SetCustomOnResults(SearchParams::TOnResults const & customOnResults)
```
Comment (ygorshenin) [inline search/search_tests_support/test_search_request.cpp]: The implementation should be after Run.

File: search/search_tests_support/test_search_request.hpp [hunk 3]
```diff
@@ -28,8 +28,19 @@ class TestSearchRequest
                     Mode mode, m2::RectD const & viewport);
   TestSearchRequest(TestSearchEngine & engine, SearchParams params, m2::RectD const & viewport);
 
+  // Initiates the search.
+  void Start();
```
Comment (ygorshenin) [inline search/search_tests_support/test_search_request.hpp]: Do we need Start() and Wait() in public section now?

Comment (mpimenov) [inline search/search_tests_support/test_search_request.hpp]: done

File: search/search_tests_support/test_search_request.cpp (hunk 1 above)
Comment (mpimenov) [inline search/search_tests_support/test_search_request.cpp]: Original OnResults notifies the condvar.

File: search/search_tests_support/test_search_request.cpp (hunk 2 above)
Comment (mpimenov) [inline search/search_tests_support/test_search_request.cpp]: done

**Pull request merged**